src/sinks/nats.rs

```diff
@@ -136,16 +143,20 @@ impl NatsSink {
 
 impl From<NatsOptions> for async_nats::Options {
     fn from(options: NatsOptions) -> Self {
-        async_nats::Options::new()
-            .with_name(&options.connection_name)
-            .reconnect_buffer_size(0)
+        (match &options.auth {
+            None => async_nats::Options::new(),
+            Some(auth) => auth.to_nats_options(),
+        })
```

Comment:
This looks like a common pattern, consider moving it into a function in `src/nats.rs`

Reply:
Actually with the newly added NATS sink, I think there's a fair bit more common code that could move into `src/nats.rs`. Shall I try moving more stuff as part of this PR, or do it in a subsequent one?

(De-duping this particular fragment from the 3 places it's happening I will do regardless)

Reply:
If it's not too large a change, then include it. Otherwise we would be happy to see another pull that reduces that tech debt.